tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -217,6 +230,33 @@ impl ConnectionWorkersScheduler {
         cancel: CancellationToken,
     ) -> Result<Self, ConnectionWorkersSchedulerError> {
         let endpoint = Self::setup_endpoint(bind, stake_identity)?;
+
+        let connection_gen = Arc::new(AtomicU64::new(0));
+        tokio::spawn({
+            let cancel = cancel.clone();
+            let connection_gen = connection_gen.clone();
+            let mut endpoint = endpoint.clone();
+            async move {
+                loop {
+                    tokio::select! {
+                        _ = update_certificate_receiver.changed() => {
+                            let stake_identity = update_certificate_receiver.borrow_and_update();
+                            let client_certificate = match stake_identity.as_ref() {
+                                Some(identity) => &identity.as_certificate(),
+                                None => &QuicClientCertificate::new(None),
+                            };
+
+                            let client_config = create_client_config(client_certificate);
+                            endpoint.set_default_client_config(client_config);
```

Comment:
I do not think this will correctly update the default config used by the worker.
Just above this async block, you cloned the `endpoint` value.
And the default client config field is not shared: https://docs.rs/quinn/0.11.7/src/quinn/endpoint.rs.html#47-51

Reply:
Worker are invalidated when the endpoint config is updated because the have access only to the connections created with the previous endpoint. So all the workers of with old certificate will be stopped.